packages/gatsby-transformer-react-docgen-typescript/gatsby-node.js

```diff
@@ -110,17 +103,72 @@ async function onCreateNode({ node, actions, loadNodeContent, createNodeId, crea
     // console.warn('No component found in', node.absolutePath);
   }
 
-  (parsedComponents || [])
-    .filter(parsed => parsed && parsed.displayName) // TabContent.tsx is being a pain so check for parsed.displayName
+  return (parsedComponents || []).filter(parsed => parsed && parsed.displayName);
+}
+
+function getInterfaceMetadata(fileNode, sourceText) {
+
+  const ast = ts.createSourceFile(
+    fileNode.absolutePath,   // fileName
+    sourceText,
+    ts.ScriptTarget.Latest // languageVersion
+  );
+  
+  function getText(node) {
+    if (!node || !node.pos || !node.end) {
+      return undefined;
+    }
+    return sourceText.substring(node.pos, node.end).trim();
+  }
+
+  const interfaces = [];
+  
+  ast.statements
+    .filter(statement => statement.kind === ts.SyntaxKind.InterfaceDeclaration)
+    .forEach(statement => {
+      const props = statement.members.map(member => {
+        return {
+          name: (member.name && member.name.escapedText) || member.parameters && `[${getText(member.parameters[0])}]` || 'Unknown',
+          description: member.jsDoc
+            ? member.jsDoc.map(doc => doc.comment).join('\n')
+            : null,
+          required: member.questionToken === undefined,
+          tsType: {
+            raw: getText(member.type).trim()
+          }
+        };
+      });
+
+      interfaces.push({
+        displayName: statement.name.escapedText,
+        props
+      });
+    });
+    return interfaces;
+}
+
+// Docs https://www.gatsbyjs.org/docs/actions/#createNode
+async function onCreateNode({ node, actions, loadNodeContent, createNodeId, createContentDigest }) {
+  if (!isComponentCode(node)) return;
+
+  const sourceText = await loadNodeContent(node);
+  const componentMeta = getComponentMetadata(node, sourceText);
+  const interfaceMeta = getInterfaceMetadata(node, sourceText);
```

Comment:
Change to `const interfaceMeta = isTSX(node) ? getInterfaceMetadata(node, sourceText) : [];`